src/Teams/Services/ManageTeamsMembersService.cs

```diff
@@ -12,18 +12,43 @@ namespace Teams.Services
 {
     public class ManageTeamsMembersService : IManageTeamsMembersService
     {
+        private readonly ICurrentUser _currentUser;
+        private readonly IRepository<Team, int> _teamRepository;
         private readonly IRepository<TeamMember, int> _teamMemberRepository;
 
         public ManageTeamsMembersService(IRepository<TeamMember, int> teamMemberRepository)
         {
             _teamMemberRepository = teamMemberRepository;
         }
 
+        public ManageTeamsMembersService(IRepository<Team, int> teamRepository, IRepository<TeamMember, int> memberRepository, ICurrentUser currentUser)
+        {
+            _currentUser = currentUser;
+            _teamRepository = teamRepository;
+            _teamMemberRepository = memberRepository;
+        }
+
+        public async Task<bool> AddAsync(int team_id, string member_id)
+        {
+            var team = await _teamRepository.GetAll().Where(t => t.Id == team_id).Include(t => t.TeamMembers).FirstOrDefaultAsync();
+            if (team != null && team.TeamOwner == _currentUser.Current.Id()
+                && member_id != _currentUser.Current.Id() && !AlreadyInTeam(team, member_id))
+            {
+                return await _teamMemberRepository.InsertAsync(new TeamMember { TeamId = team_id, MemberId = member_id });
+            }
+            return false;
+        }
+
+        private bool AlreadyInTeam(Team team, string member_id)
+        {
+            return team.TeamMembers.Any(t => t.MemberId == member_id);
+        }
+
         public async Task<TeamMember> GetMemberAsync(int team_id, string member_id)
         {
-             return await _teamMemberRepository.GetAll()
-                    .Where(x => x.MemberId == member_id && x.TeamId == team_id)
-                    .FirstOrDefaultAsync();
+            return await _teamMemberRepository.GetAll()
```

Comment:
You are already using include to get all team members in AddAsync method, so this means that you can use this check over collection that is already stored in memory, the second approach if you do not want to actually retrieve all information you can use check on sql level to avoid unnecessary load. Something like AnyAsync(team => team.Id = team_id || team.TeamMembers.Any(entry => entry.Member == member_id && entry.TeamId == team_id))